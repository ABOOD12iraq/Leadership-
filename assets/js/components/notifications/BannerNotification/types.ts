/**
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export type LEARN_MORE_TARGET = 'external' | 'internal';

export type BannerNotificationProps = {
	id: string;
	className?: string;
	children?: React.ReactNode;
	title: string;
	description?: React.ReactNode;
	learnMoreURL?: string;
	learnMoreDescription?: string;
	learnMoreLabel?: string;
	learnMoreTarget?: LEARN_MORE_TARGET;
	WinImageSVG?: React.ElementType;
	SmallImageSVG?: React.ElementType;
	format?: string;
	ctaLink?: string;
	ctaLabel?: string;
	ctaTarget?: string;
	type?: string;
	dismiss?: string;
	isDismissible?: boolean;
	logo?: boolean;
	module?: string;
	moduleName?: string;
	dismissExpires?: number;
	showOnce?: boolean;
	onCTAClick?: ( event: React.MouseEvent ) => void | Promise< any >;
	onView?: () => void;
	onDismiss?: ( event: React.MouseEvent ) => void | Promise< any >;
	onLearnMoreClick?: ( event: React.MouseEvent ) => void | Promise< any >;
	badgeLabel?: string;
	rounded?: boolean;
	footer?: React.ReactNode;
	secondaryPane?: React.ReactNode;
	showSmallWinImage?: boolean;
	smallWinImageSVGWidth?: number;
	smallWinImageSVGHeight?: number;
	mediumWinImageSVGWidth?: number;
	mediumWinImageSVGHeight?: number;
	ctaComponent?: React.ReactNode;
};

// eslint-disable-next-line sitekit/acronym-case
export type BannerProps = React.HTMLProps< HTMLElement > & {
	id?: string;
	secondaryPane?: React.ReactNode;
};

export type BannerActionsProps = {
	ctaLink?: string;
	ctaLabel?: string;
	ctaComponent?: React.ReactElement;
	ctaTarget?: string;
	ctaCallback?: ( ...args: any[] ) => Promise< any > | void;
	dismissLabel?: string;
	dismissCallback?: ( ...args: any[] ) => Promise< any > | void;
};

export type BannerDescriptionProps = {
	description: React.ReactNode;
	learnMoreLabel?: string;
	learnMoreURL?: string;
	learnMoreDescription?: string;
	learnMoreTarget?: LEARN_MORE_TARGET;
	onLearnMoreClick?: () => void;
};

export type BannerIconProps = {
	type?: string;
};
