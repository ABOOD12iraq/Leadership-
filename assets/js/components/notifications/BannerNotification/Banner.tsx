/**
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import classnames from 'classnames';

/**
 * WordPress dependencies
 */
import { Fragment, forwardRef } from '@wordpress/element';

/*
 * Internal dependencies
 */
import { Cell, Grid, Row } from '../../../material-components';
import { BannerProps } from './types';

// eslint-disable-next-line sitekit/acronym-case
const Banner = forwardRef< HTMLElement, BannerProps >(
	( { id, className, children, secondaryPane }, ref ) => (
		<section
			id={ id }
			className={ classnames( className, 'googlesitekit-publisher-win' ) }
			ref={ ref }
		>
			<Grid>
				<Row>{ children }</Row>
			</Grid>

			{ secondaryPane && (
				<Fragment>
					<div className="googlesitekit-publisher-win__secondary-pane-divider" />
					<Grid className="googlesitekit-publisher-win__secondary-pane">
						<Row>
							<Cell
								className="googlesitekit-publisher-win__secondary-pane"
								size={ 12 }
							>
								{ secondaryPane }
							</Cell>
						</Row>
					</Grid>
				</Fragment>
			) }
		</section>
	)
);

Banner.displayName = 'Banner';

export default Banner;
