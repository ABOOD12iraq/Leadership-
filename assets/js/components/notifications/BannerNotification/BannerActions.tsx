/**
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import { useMountedState } from 'react-use';

/*
 * WordPress dependencies
 */
import { useState } from '@wordpress/element';

/*
 * Internal dependencies
 */
import Data from 'googlesitekit-data';
import { Button, SpinnerButton } from 'googlesitekit-components';
import { CORE_LOCATION } from '../../../googlesitekit/datastore/location/constants';
import Link from '../../Link';
const { useSelect } = Data;

interface BannerActionsProps {
	ctaLink?: string;
	ctaLabel?: string;
	ctaComponent?: React.ReactElement;
	ctaTarget?: string;
	ctaCallback?: ( ...args: any[] ) => Promise< any > | void;
	dismissLabel?: string;
	dismissCallback?: ( ...args: any[] ) => Promise< any > | void;
}

export default function BannerActions( props: BannerActionsProps ) {
	const {
		ctaLink,
		ctaLabel,
		ctaComponent,
		ctaTarget,
		ctaCallback,
		dismissLabel,
		dismissCallback,
	} = props;

	const [ isAwaitingCTAResponse, setIsAwaitingCTAResponse ] =
		useState( false );

	const isMounted = useMountedState();

	const isNavigatingToCTALink = useSelect( ( select ) =>
		ctaLink ? select( CORE_LOCATION ).isNavigatingTo( ctaLink ) : false
	) as boolean;

	const handleCTAClick = async ( ...params ) => {
		setIsAwaitingCTAResponse( true );
		await ctaCallback?.( ...params );
		if ( isMounted() ) {
			setIsAwaitingCTAResponse( false );
		}
	};

	if ( ! ctaLink && ! dismissLabel && ! ctaComponent ) {
		return null;
	}

	// ctaLink links are always buttons, in which case the dismiss should be a Link.
	// If there is only a dismiss however, it should be the primary action with a Button.
	const DismissComponent = ctaLink || ctaComponent ? Link : Button;

	return (
		<div className="googlesitekit-publisher-win__actions">
			{ ctaComponent }

			{ ctaLabel && (
				<SpinnerButton
					className="googlesitekit-notification__cta"
					href={ ctaLink }
					target={ ctaTarget }
					onClick={ handleCTAClick }
					disabled={ isAwaitingCTAResponse || isNavigatingToCTALink }
					isSaving={ isAwaitingCTAResponse || isNavigatingToCTALink }
				>
					{ ctaLabel }
				</SpinnerButton>
			) }

			{ dismissLabel && (
				<DismissComponent
					onClick={ dismissCallback }
					disabled={ isAwaitingCTAResponse || isNavigatingToCTALink }
				>
					{ dismissLabel }
				</DismissComponent>
			) }
		</div>
	);
}
