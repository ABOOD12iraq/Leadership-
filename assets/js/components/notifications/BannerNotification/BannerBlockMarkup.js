/**
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';
import { map } from 'lodash';

/*
 * Internal dependencies
 */
import { Grid, Cell, Row } from '../../../material-components';
import DataBlock from '../../DataBlock';

export default function BannerBlockMarkup( props ) {
	const { blockData, inlineLayout } = props;

	if ( ! blockData ) {
		return null;
	}

	return (
		<Grid>
			<Row>
				{ map( blockData, ( block, i ) => (
					<Cell key={ i } lgSize={ inlineLayout ? 5 : 4 }>
						<div className="googlesitekit-publisher-win__stats">
							<DataBlock { ...block } />
						</div>
					</Cell>
				) ) }
			</Row>
		</Grid>
	);
}

BannerBlockMarkup.propTypes = {
	inlineLayout: PropTypes.bool,
	blockData: PropTypes.array,
};
