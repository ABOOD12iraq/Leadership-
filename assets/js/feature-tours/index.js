/**
 * Feature Tours.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { VIEW_CONTEXT_DASHBOARD } from '../googlesitekit/constants';
import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';

const allTrafficWidget = {
	slug: 'allTrafficWidget',
	contexts: [ VIEW_CONTEXT_DASHBOARD ],
	version: '1.25.0',
	checkRequirements: async ( registry ) => {
		// Here we need to wait for the underlying selector to be resolved before selecting `isModuleConnected`.
		await registry.__experimentalResolveSelect( CORE_MODULES ).getModules();
		if ( ! registry.select( CORE_MODULES ).isModuleConnected( 'analytics' ) ) {
			return false;
		}
		return true;
	},
	steps: [
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__user-count-chart',
			title: __( "It's now easier to see your site's traffic at a glance", 'google-site-kit' ),
			content: __( 'Check the trend graph to see how your traffic changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-widget--analyticsAllTraffic__dimensions-chart',
			title: __( 'See where your visitors come from', 'google-site-kit' ),
			content: __( 'Click on the chart slices to see how each segment has changed over time', 'google-site-kit' ),
		},
		{
			target: '.googlesitekit-header__date-range-selector-menu',
			title: __( 'Check how your traffic changed since you last looked', 'google-site-kit' ),
			content: __( 'Select a time frame to see the comparison with the previous time period', 'google-site-kit' ),
		},
	],
};

// Ordered tours.
export default [
	allTrafficWidget,
];
