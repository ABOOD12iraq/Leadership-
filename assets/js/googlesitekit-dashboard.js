/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { clearWebStorage } from './util';
import Root from './components/Root';
import './components/legacy-notifications';
import {
	VIEW_CONTEXT_DASHBOARD,
	VIEW_CONTEXT_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_MODULE_SETUP,
} from './googlesitekit/constants';
import DashboardEntryPoint from './components/DashboardEntryPoint';

// Initialize the app once the DOM is ready.
domReady( async () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		clearWebStorage();
	}

	const renderTarget = document.getElementById(
		'js-googlesitekit-dashboard'
	);

	if ( renderTarget ) {
		// Make a separate API request since we cannot use selectors
		// outside the Root component.
		const { authenticated: isAuthenticated } = await API.get(
			'core',
			'user',
			'authentication',
			{},
			{ useCache: false }
		);

		const { setupModuleSlug } = renderTarget.dataset;
		let viewContext = VIEW_CONTEXT_MODULE_SETUP;
		if ( ! setupModuleSlug ) {
			viewContext = isAuthenticated
				? VIEW_CONTEXT_DASHBOARD
				: VIEW_CONTEXT_DASHBOARD_VIEW_ONLY;
		}
		render(
			<Root viewContext={ viewContext }>
				<DashboardEntryPoint setupModuleSlug={ setupModuleSlug } />
			</Root>,
			renderTarget
		);
	}
} );
