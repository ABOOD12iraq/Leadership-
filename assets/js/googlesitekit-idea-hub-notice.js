/**
 * Idea Hub Notice (for Block Editor) entrypoint.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __, sprintf } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { getItem, setItem } from './googlesitekit/api/cache';

const shownNotices = [];

const editorNoticeKey = ( postID ) => {
	return `modules::idea-hub::dismissed-editor-notice-${ postID }`;
};

const loadIdeaHubNotices = async ( _global = global ) => {
	const { wp } = _global;

	const hasNotice = ( postID ) => {
		if ( wp.data.select( 'core/notices' ).getNotices() === undefined ) {
			return undefined;
		}

		return wp.data.select( 'core/notices' ).getNotices().some( ( notice ) => {
			return notice.id === editorNoticeKey( postID );
		} );
	};

	const listener = async () => {
		// eslint-disable-next-line sitekit/acronym-case
		const postID = wp.data.select( 'core/editor' ).getCurrentPostId();

		if ( ! postID ) {
			return;
		}

		const { cacheHit } = await getItem( editorNoticeKey( postID ) );

		if ( hasNotice( postID ) === false ) {
			// We've already shown this notice, so when it's hidden, mark it as shown
			// so it doesn't appear again.
			if ( shownNotices.includes( editorNoticeKey( postID ) ) ) {
				setItem(
					editorNoticeKey( postID ),
					postMeta.googlesitekitpersistent_idea_text,
					{ ttl: 108000 } // Don't show this notice for another 90 days.
				);
				unsubscribeFromListener();
				return;
			}

			// We've already shown this notice on a previous visit to this page in
			// the editor, so don't show it again.
			if ( cacheHit ) {
				unsubscribeFromListener();
				return;
			}

			// We haven't shown any notice for this post before, so let's check for
			// Idea Hub postmeta.
			const postMeta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );

			// eslint-disable-next-line camelcase
			if ( postMeta?.googlesitekitpersistent_idea_text ) {
				wp.data.dispatch( 'core/notices' ).createInfoNotice(
					/* translators: %s: Idea post name */
					sprintf( __( 'This post was created from an idea you picked in Site Kitâ€™s Idea Hub: %s', 'google-site-kit' ), postMeta.googlesitekitpersistent_idea_text ),
					{ id: editorNoticeKey( postID ) }
				);

				// Mark the notice as shown locally, so if it is no longer marked as
				// visible, we known it was dismissed (rather than having yet to be
				// displayed).
				shownNotices.push( editorNoticeKey( postID ) );
			}
		}
	};

	const unsubscribeFromListener = wp.data.subscribe( listener );
};

loadIdeaHubNotices();
