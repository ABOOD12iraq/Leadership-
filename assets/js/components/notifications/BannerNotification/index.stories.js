/**
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import AwardSVG from '../../../../svg/graphics/award.svg';
import BannerNotification from './index';

function Template( { ...args } ) {
	return <BannerNotification { ...args } />;
}

export const SmallWithImage = Template.bind( {} );
SmallWithImage.storyName = 'Small with Image';
SmallWithImage.args = {
	id: 'notification-id',
	title: 'Congrats on your first post!',
	description: 'We sent your sitemap to Googlebot.',
	learnMore: 'Learn more',
	dismiss: 'OK, Got it!',
	format: 'small',
	SmallImageSVG: AwardSVG,
	type: 'win-success',
};

export const SmallWithNoImage = Template.bind( {} );
SmallWithNoImage.storyName = 'Small with No Image';
SmallWithNoImage.args = {
	id: 'notification-id',
	title: 'Your Site is Now Registered!',
	description: 'Congrats your site is registered with Search Console!',
	dismiss: 'OK, Got it!',
	learnMoreURL: 'http://google.com',
	learnMoreLabel: 'Learn more',
	learnMoreDescription: 'about the particular win',
	format: 'small',
	type: 'win-success',
};

export const SmallWithError = Template.bind( {} );
SmallWithError.storyName = 'Small with Error';
SmallWithError.args = {
	id: 'notification-id',
	title: 'AMP Validation Error',
	description: 'There are validation errors that need to be fixed.',
	learnMoreURL: 'http://google.com',
	learnMore: 'View Search Console report',
	learnMoreLabel: 'View Search Console',
	learnMoreDescription: 'report',
	dismiss: 'Dismiss',
	format: 'small',
	type: 'win-error',
};

export const SmallWithWarnign = Template.bind( {} );
SmallWithWarnign.storyName = 'Small with Warning';
SmallWithWarnign.args = {
	id: 'notification-id',
	title: 'Index Warning',
	description: 'Indexed, though blocked by robots.txt.',
	learnMoreURL: 'http://google.com',
	learnMoreLabel: 'Learn more',
	dismiss: 'Dismiss',
	format: 'small',
	ctaLink: 'http://google.com',
	ctaLabel: 'Validate',
	type: 'win-warning',
};

export const SmallWithElementDescription = Template.bind( {} );
SmallWithElementDescription.storyName = 'Small with Element Description';
SmallWithElementDescription.args = {
	id: 'notification-id',
	title: 'Index Warning',
	description: <p>This description is a React element.</p>,
	learnMoreURL: 'http://google.com',
	learnMoreLabel: 'Learn more',
	dismiss: 'Dismiss',
	format: 'small',
	ctaLink: 'http://google.com',
	ctaLabel: 'Validate',
	type: 'win-warning',
};

export default {
	title: 'Components/BannerNotification',
	component: BannerNotification,
};
