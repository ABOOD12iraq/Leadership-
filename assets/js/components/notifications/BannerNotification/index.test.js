/**
 * BannerNotification component tests.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import BannerNotification from './index';
import { render } from '../../../../../tests/js/test-utils';

describe( 'BannerNotification', () => {
	it( 'should render span with dangerouslySetInnerHTML', () => {
		const { container, getByText } = render(
			<BannerNotification
				id="fake"
				title={ 'Hey there!' }
				description={ 'I am string, not React elemet' }
			/>
		);

		expect(
			container.querySelector(
				'.googlesitekit-publisher-win__inner-html'
			)
		).toBeInTheDocument();
		expect(
			getByText( /I am string, not React elemet/ )
		).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );

	it( 'should render React element', () => {
		const { container } = render(
			<BannerNotification
				id="fake"
				title={ 'Hey there!' }
				description={
					<p className="sk-react-element">I am React elemet</p>
				}
			/>
		);

		expect(
			container.querySelector(
				'.googlesitekit-publisher-win__inner-html'
			)
		).toBeFalsy();

		expect(
			container.querySelector( '.sk-react-element' )
		).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );
} );
