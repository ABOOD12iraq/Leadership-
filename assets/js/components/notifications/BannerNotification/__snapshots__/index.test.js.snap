// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerNotification attribute description with text should render React element 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-publisher-win--is-open"
    id="fake"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-12-desktop mdc-layout-grid__cell--span-8-tablet mdc-layout-grid__cell--order-1-tablet mdc-layout-grid__cell--span-4-phone mdc-layout-grid__cell--order-2-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            Hey there!
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p
              class="sk-react-element"
            >
              I am React elemet
            </p>
          </div>
          <div
            class="googlesitekit-publisher-win__actions"
          >
            <span
              class="spinner"
              style="display: none; float: none; margin-top: 0px; visibility: visible;"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`BannerNotification attribute description with text should render span with dangerouslySetInnerHTML 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-publisher-win--is-open"
    id="fake"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-12-desktop mdc-layout-grid__cell--span-8-tablet mdc-layout-grid__cell--order-1-tablet mdc-layout-grid__cell--span-4-phone mdc-layout-grid__cell--order-2-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            Hey there!
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p>
              <span
                class="googlesitekit-publisher-win__inner-html"
              >
                I am string, not React elemet
              </span>
            </p>
          </div>
          <div
            class="googlesitekit-publisher-win__actions"
          >
            <span
              class="spinner"
              style="display: none; float: none; margin-top: 0px; visibility: visible;"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`BannerNotification should render React element 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-publisher-win--is-open"
    id="fake"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-12-desktop mdc-layout-grid__cell--span-8-tablet mdc-layout-grid__cell--order-1-tablet mdc-layout-grid__cell--span-4-phone mdc-layout-grid__cell--order-2-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            Hey there!
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p
              class="sk-react-element"
            >
              I am React elemet
            </p>
          </div>
          <div
            class="googlesitekit-publisher-win__actions"
          >
            <span
              class="spinner"
              style="display: none; float: none; margin-top: 0px; visibility: visible;"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`BannerNotification should render span with dangerouslySetInnerHTML 1`] = `
<div>
  <section
    class="googlesitekit-publisher-win googlesitekit-publisher-win--is-open"
    id="fake"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-publisher-win__content mdc-layout-grid__cell mdc-layout-grid__cell--span-12-desktop mdc-layout-grid__cell--span-8-tablet mdc-layout-grid__cell--order-1-tablet mdc-layout-grid__cell--span-4-phone mdc-layout-grid__cell--order-2-phone"
        >
          <h3
            class="googlesitekit-heading-2 googlesitekit-publisher-win__title"
          >
            Hey there!
          </h3>
          <div
            class="googlesitekit-publisher-win__desc"
          >
            <p>
              <span
                class="googlesitekit-publisher-win__inner-html"
              >
                I am string, not React elemet
              </span>
            </p>
          </div>
          <div
            class="googlesitekit-publisher-win__actions"
          >
            <span
              class="spinner"
              style="display: none; float: none; margin-top: 0px; visibility: visible;"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
